qiskit/algorithms/evolvers/evolution_problem.py

```diff
@@ -44,15 +44,92 @@ def __init__(
             initial_state: Quantum state to be evolved.
             aux_operators: Optional list of auxiliary operators to be evaluated with the
                 evolved ``initial_state`` and their expectation values returned.
+            truncation_threshold: Defines a threshold under which values can be assumed to be 0.
+                Used when ``aux_operators`` is provided.
             t_param: Time parameter in case of a time-dependent Hamiltonian. This
                 free parameter must be within the ``hamiltonian``.
             hamiltonian_value_dict: If the Hamiltonian contains free parameters, this
                 dictionary maps all these parameters to values.
+
+        Raises:
+            ValueError: If non-positive time of evolution is provided.
+            ValueError: If no ``initial_state`` is provided.
+            ValueError: If not all parameter values are provided.
         """
 
+        self.t_param = t_param
+        self.hamiltonian_value_dict = hamiltonian_value_dict
         self.hamiltonian = hamiltonian
         self.time = time
         self.initial_state = initial_state
         self.aux_operators = aux_operators
-        self.t_param = t_param
-        self.hamiltonian_value_dict = hamiltonian_value_dict
+        self.truncation_threshold = truncation_threshold
+
+    @property
+    def hamiltonian(self) -> OperatorBase:
```

Comment:
Right now we have trivial setters for some attributes but not all. Do we want them for all the attributes, like `aux_operators` or `hamiltonian_value_dict` too? Or we could also delete all trivial ones (would be my slight preference! 🙂)